spec/integration/rails/finders_spec.rb

```diff
@@ -383,58 +393,58 @@ def json
-          #fields: { houses: 'name', condos: 'condo_description' },
-          #extra_fields: { houses: 'house_price', condos: 'condo_price' }
-        #}
-        #house = json_includes('houses')[0]['attributes']
-        #condo = json_includes('condos')[0]['attributes']
-        #expect(house).to have_key('name')
-        #expect(house).to have_key('house_price')
-        #expect(house).to_not have_key('house_description')
-        #expect(condo).to have_key('condo_description')
-        #expect(condo).to have_key('condo_price')
-        #expect(condo).to_not have_key('name')
-      #end
-
-      #it 'allows additional levels of nesting' do
-        #get :index, params: { include: 'dwelling.state' }
-        #expect(json_includes('states').length).to eq(1)
-      #end
-    #end
+    context 'sideloading polymorphic belongs_to' do
+      it 'allows extra fields for the sideloaded resource' do
+        get :index, params: {
+          include: 'dwelling',
+          extra_fields: { houses: 'house_price', condos: 'condo_price' }
+        }
+        house = json_includes('houses')[0]['attributes']
+        expect(house['name']).to be_present
+        expect(house['house_description']).to be_present
+        expect(house['house_price']).to eq(1_000_000)
+        condo = json_includes('condos')[0]['attributes']
+        expect(condo['name']).to be_present
+        expect(condo['condo_description']).to be_present
+        expect(condo['condo_price']).to eq(500_000)
+      end
+
+      it 'allows sparse fieldsets for the sideloaded resource' do
+        get :index, params: {
+          include: 'dwelling',
+          fields: { houses: 'name', condos: 'condo_description' }
+        }
+        house = json_includes('houses')[0]['attributes']
+        expect(house['name']).to be_present
+        expect(house).to_not have_key('house_description')
+        expect(house).to_not have_key('house_price')
+        condo = json_includes('condos')[0]['attributes']
+        expect(condo['condo_description']).to be_present
+        expect(condo).to_not have_key('name')
+        expect(condo).to_not have_key('condo_price')
+      end
+
+      it 'allows extra fields and sparse fieldsets for the sideloaded resource' do
+        get :index, params: {
+          include: 'dwelling',
+          fields: { houses: 'name', condos: 'condo_description' },
+          extra_fields: { houses: 'house_price', condos: 'condo_price' }
+        }
+        house = json_includes('houses')[0]['attributes']
+        condo = json_includes('condos')[0]['attributes']
+        expect(house).to have_key('name')
+        expect(house).to have_key('house_price')
+        expect(house).to_not have_key('house_description')
+        expect(condo).to have_key('condo_description')
+        expect(condo).to have_key('condo_price')
+        expect(condo).to_not have_key('name')
+      end
+
+      it 'allows additional levels of nesting' do
+        get :index, params: { include: 'dwelling.state' }
```

Comment:
Does this work if only one of Condo or House have a state?

Reply:
I was about to reply [yes](https://github.com/richmolj/jsonapi_compliable/blob/poly_belongs_to/spec/polymorphism_spec.rb#L77-L101) but just tried it and no. Will address this, great catch.